auth.go

```diff
@@ -59,6 +61,10 @@ func pickSignatureAlgorithm(pubkey crypto.PublicKey, peerSigAlgs, ourSigAlgs []S
 			if sigType == signatureECDSA {
 				return sigAlg, sigType, hashAlg, nil
 			}
+		case ed25519.PublicKey:
+			if sigType == signatureEd25519 {
```

Comment:
Note: besides TLS 1.3, this can also select Ed25519 for TLS 1.2 and is defined in:
https://tools.ietf.org/html/draft-ietf-tls-rfc4492bis